src/librustc_interface/interface.rs

```diff
@@ -80,6 +81,8 @@ pub struct Config {
 
     pub crate_name: Option<String>,
     pub lint_caps: FxHashMap<lint::LintId, lint::Level>,
+
+    pub register_lints: Option<Box<dyn Fn(&Session, &mut lint::LintStore) + Send + Sync>>,
```

Comment:
It'd be nice to have a comment here. Something like:

---

Callback that is invoked during lint registration. Use the `&mut LintStore` value to register new lints with the compiler as needed.